File: src/main/java/com/github/urlshorten/Url.java
package com.github.urlshorten;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
public class Url {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "original_url", length = 2083, nullable = false)
    private String originalUrl;

    @Column(name = "shorten_url", length = 62)
    private String shortenUrl;

    @Column(name = "creation_date", nullable = false)
    private LocalDateTime creationDate;

    @Column(name = "expiration_date",  nullable = false)
    private LocalDateTime expirationDate;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getOriginalUrl() {
        return originalUrl;
    }

    public void setOriginalUrl(String originalUrl) {
        this.originalUrl = originalUrl;
    }

    public String getShortenUrl() {
        return shortenUrl;
    }

    public void setShortenUrl(String shortenUrl) {
        this.shortenUrl = shortenUrl;
    }

    public LocalDateTime getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDateTime creationDate) {
        this.creationDate = creationDate;
    }

    public LocalDateTime getExpirationDate() {
        return expirationDate;
    }

    public void setExpirationDate(LocalDateTime expirationDate) {
        this.expirationDate = expirationDate;
    }
}

File: src/main/kotlin/com/github/urlshorten/Url.kt
package com.github.urlshorten

import java.time.LocalDateTime
import javax.persistence.*

@Entity
class Url(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long?,

        @Column(name = "original_url", length = 2083, nullable = false)
        var originalUrl: String,

        @Column(name = "shorten_url", length = 62)
        var shortenUrl: String?,

        @Column(name = "creation_date", nullable = false)
        var creationDate: LocalDateTime,

        @Column(name = "expiration_date", nullable = false)
        var expirationDate: LocalDateTime) {}